module/VuFind/src/VuFind/Controller/AbstractBase.php

```diff
@@ -164,6 +174,27 @@ protected function createViewModel($params = null)
         return new ViewModel($params);
     }
 
+    /**
+     * Execute the request
+     *
+     * @param MvcEvent $e MVC event object
+     *
+     * @return mixed
+     *
+     * @throws Exception\DomainException
+     */
+    public function onDispatch(MvcEvent $e)
+    {
+        $response = $this->getResponse();
+        $headers = $response->getHeaders();
+        $headers->addHeaderLine(
+            'Content-Security-Policy',
+            "script-src 'strict-dynamic' 'nonce-$this->nonce' 'unsafe-inline' "
```

Comment:
Do we eventually want to make some or all of this configurable?

Reply:
If it's too complex to put in an .ini file due to dynamic elements like the nonce, maybe there should be a service that builds the string so it's easier to override.

Reply:
I agree that some service which could generate the header would be fine, maybe we can even find some library, which could make it for us... like https://github.com/paragonie/csp-builder but maybe this library is way too complex for us...

Reply:
I added the todo item for this. Some kind of flexibility does make sense here...

Reply:
The csp-builder library certainly looks like a possibility -- I'd rather use a well-established library than build something new, as long as the library meets our needs and doesn't overcomplicate things too much. In this case, it looks like we have the option of using a config file to initialize the CSP, or we could just build a factory to do the work programmatically.

Reply:
I agree, I looked into other library too: https://github.com/BePsvPT/secure-headers, but it looks even more complex then paragonie/csp-builder, so I think we can give a try to it.
 
Just one question remains here: If it is better to do in configuration, or programatically...

If we go with configuration, it will probably have more flexibility for non-programer implementers, but it is in JSON, and I think we can't use configuration inheritence then... So there will be a bit of inconsistence with other configuration.

Reply:
I think if we load the configuration through the factory, then we can address both scenarios: by default, we can load a configuration, but a user could override the factory to do a programmatic configuration instead if they wanted to. You're right that JSON limits our ability to allow configurations to inherit from one another, but we can at least allow file overriding through the local directory by selecting the file using `\VuFind\Config\Locator::getConfigPath()`.